src/plone/restapi/serializer/dxcontent.py

```diff
@@ -84,6 +86,24 @@ def __call__(self, version=None, include_items=True):
                 value = serializer()
                 result[json_compatible(name)] = value
 
+        # Check if discussion is allowed on the object
+        # The original code and condition worflow is in
+        # https://github.com/plone/plone.app.discussion/blob/master/plone/app/discussion/browser/conversation.py#L111
+        registry = queryUtility(IRegistry)
+        settings = registry.forInterface(IDiscussionSettings, check=False)
+
+        # Check if discussion is allowed globally, overwritting whatever that
+        # is set in the object
+        if not settings.globally_enabled:
+            result['allow_discussion'] = False
```

Comment:
@sneridagh What happens if there is no registry data for IDiscussionSettings? Does settings.globally_enabled fail?

Reply:
I guess p.a.discussion will be around in the current form and is not going anywhere for a long time, right? :P

Reply:
I was just wondering if we should remove the 'check=False' bit just above it :)

Reply:
Why do we use check=False in the first place? Wouldn't it be better to fail hard in case p.a.discussion is not around? The other option is to really make it optional by checking if "setting != None" before proceeding...

Reply:
Removed check=False